src/nopoll_conn.c

```diff
@@ -3463,7 +3463,7 @@ noPollMsg   * nopoll_conn_get_msg (noPollConn * conn)
 read_payload:
 
 	/* copy payload received */
-	msg->payload = nopoll_new (char, msg->payload_size + 1);
+	msg->payload = nopoll_new (char, msg->payload_size + 1);	/* allow extra byte for string terminator */
```

Comment:
I think the `+1` is a bug since it appears that message payloads are either binary blobs (not NIL terminated) or UTF-8 characters (also not explicitly NIL terminated).

Reply:
In principle I would agree with you. However, I didn't want to break anything that was already relying on the terminator that is already being put there (currently by `__nopoll_conn_receive()`). One example being the `strdup` of the reason payload in a connection close.